src/collector/types.ts

```diff
@@ -14,18 +12,89 @@ export interface PaginatedResponse {
   total: number;
 }
 
-export interface CiscoAmpApiResponse {
+/**
+ * CiscoAmpApiResponse structure
+ */
+export interface CiscoAmpApiResponse<T> {
   version: string;
   metadata: {
     links: {
       self: string;
     };
     results: PaginatedResponse;
   };
+  data: T[] | T;
 }
 
-export interface ListComputersResponse extends CiscoAmpApiResponse {
-  data: CiscoAmpComputer[];
+/**
+ * CiscoAmpComputer resource  structure
+ */
+export interface CiscoAmpComputer {
+  connector_guid: string;
+  hostname: string;
+  windows_processor_id: string;
+  active: boolean;
+  links: {
+    computer: string;
+    trajectory: string;
+    group: string;
+  };
+  connector_version: string;
+  operating_system: string;
+  internal_ips: string[];
+  external_ip: string;
+  group_guid: string;
+  install_date: string;
+  is_compromised: boolean;
+  demo: boolean;
+  network_addresses: { mac: string; ip: string }[];
+  policy: {
+    guid: string;
+    name: string;
+  };
+  groups: { guid: string; name: string }[];
+  last_seen: string;
+  faults: any[];
+  isolation: {
+    available: boolean;
+    status: string;
+  };
+  orbital: {
+    status: string;
+  };
 }
 
-export type CiscoAmpComputer = Opaque<any, 'CiscoAmpComputer'>;
+export interface CiscoAmpVulnerability {
+  application: string;
+  version: string;
+  file: {
+    filename: string;
+    identity: {
+      sha256: string;
+    };
+  };
+  cves: {
+    id: string;
+    link: string;
+    cvss: number;
+  }[];
+  latest_timestamp: number;
+  latest_date: string;
+  groups: {
+    name: string;
+    description: string;
+    guid: string;
+  }[];
+  computers_total_count: number;
+  computers?: {
```

Comment:
Would we be able to use this computer object to create a relationship between computers and vulnerabilities?  100% understand if that's outside of the scope of this task.